docs/DeveloperGuide.md

```diff
@@ -2,37 +2,527 @@
 
-{Give non-functional requirements}
+* Should work on any _mainstream_ OS as long as it has Java `17` or above installed.
+* Should support future question types without major architectural changes.
+* Text output should be readable in standard terminals (no reliance on special colours or fonts e.g. Nerd fonts)
+* Programs should train users' understanding and memory of concepts (default list command should not give away answers,
+MCQ should not be the same options tagged to the same answers each time, etc.).
+* Error messages should be clear and actionable (e.g. Please enter a valid ___).
+* Number of questions added should not be limited by the program.
+* Commands should not be overly complicated to use, for example having to type too many items in one long single line
+command. This can be solved by breaking down the command in a user-friendly manner with multistep commands.
 
 ## Glossary
 
-* *glossary item* - Definition
+* *Multistep command* - A feature which requires the user to go through several steps to complete.
+* *FITB* - Fill in the Blanks question type, where the answer is a string entered by the user.
+* *MCQ* - Multiple Choice Question question type, where the answer is one of the options A, B, C or D.
+* *TF* - True/False question type, where the answer is a boolean true or false.
+* *FSM* - Finite State Machine, used in multistep command handling.
+* *CLI* - Command Line Interface.
+* *Command Pattern* - A behavioral design pattern that encapsulates requests as objects, allowing parameterization and 
+queuing of operations. The design pattern used for command handling.
+* *Layered Architecture* - A software design approach that organizes components into hierarchical layers, where each
+layer has a specific responsibility and interacts only with adjacent layers. The architectural style used in the design.
 
 ## Instructions for manual testing
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+### Launch and Shutdown
+1. Initial launch
+   1. Download the jar file.
+   2. Copy the file to the folder you want to use as the _home_ folder.
+   3. Open a command terminal, and `cd` into the folder you put the jar file in.
+   4. Run `java -jar MindExpander.jar` to run MindExpander.
+   5. The following screen should be displayed:
+   ```
+   ==============================
+    
+       _____  .__            .______________                                .___            
+      /     \ |__| ____    __| _/\_   _____/__  ______________    ____    __| _/___________
+     /  \ /  \|  |/    \  / __ |  |    __)_\  \/  /\____ \__  \  /    \  / __ |/ __ \_  __ \
+    /    Y    \  |   |  \/ /_/ |  |        \>    < |  |_> > __ \|   |  \/ /_/ \  ___/|  | \/
+    \____|__  /__|___|  /\____ | /_______  /__/\_ \|   __(____  /___|  /\____ |\___  >__|   
+            \/        \/      \/         \/      \/|__|       \/     \/      \/    \/
+    
+    Presented by: CS2113-F12-3
```

Comment:
This part seems to be rendering incorrectly on the actual DG. You might have to modify the markdown for this part as it uses special symbols for it to correctly render on the DG.